### PR TITLE
(wmr) - support typescript configs

### PR DESCRIPTION
**Pull request by JoviDeCroock**

### CODE REVIEW

File: packages/wmr/src/lib/normalize-options.js [hunk 1]
```diff
@@ -61,14 +62,24 @@ export async function normalizeOptions(options, mode) {
...
+		const fileUrl = url.pathToFileURL(output);
+		custom = await eval('(x => import(x))')(fileUrl);
+	} else if (hasMjsConfig || (await isFile(resolve(options.root, 'wmr.config.js')))) {
+		let initialConfigFile = hasMjsConfig ? 'wmr.config.mjs' : 'wmr.config.js',
 			initialError;
 		try {
 			const resolved = resolve(options.root, initialConfigFile);
 			const fileUrl = url.pathToFileURL(resolved);
+			const result = await compileSingleModule(resolved, { cwd: options.cwd, out: options.out });
+			console.log(result);
```
Comment (marvinhagemeister) [inline packages/wmr/src/lib/normalize-options.js:82]: Nit: console.log

File: packages/wmr/demo/wmr.config.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default function (config) {
+	return config;
```
Comment (marvinhagemeister) [inline packages/wmr/demo/wmr.config.ts:2]: Let's add a test for this to verify that the config is actually loaded.

File: packages/wmr/test/fixtures/publicpath-typescript/wmr.config.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default function (config) {
```
Comment (developit) [inline packages/wmr/test/fixtures/publicpath-typescript/wmr.config.ts:1]: I'm wondering if the `compileSingleModule` with `out='./cache'` will cause imports to be off by one directory level?

We could try doing a simple import of a second TS file here to check.

**Review by marvinhagemeister — APPROVED** [review]
Love this!

File: packages/wmr/src/lib/normalize-options.js [hunk 4]
```diff
@@ -61,10 +62,19 @@ export async function normalizeOptions(options, mode) {
...
-			initialConfigFile = hasMjsConfig ? 'wmr.config.mjs' : 'wmr.config.js',
+
+	let custom;
+	if (hasTsConfig) {
+		const resolved = resolve(options.root, 'wmr.config.ts');
+		await compileSingleModule(resolved, { cwd: options.cwd, out: resolve('.'), hmr: false });
+		const output = resolve('.', 'wmr.config.js');
+		const fileUrl = url.pathToFileURL(output);
+		custom = await eval('(x => import(x))')(fileUrl);
+		fs.unlink(resolve('.', 'wmr.config.js'));
```
Comment (danielweck) [inline packages/wmr/src/lib/normalize-options.js:75]: `resolve('.', 'wmr.config.js')` can be replaced with `output`?

**Pull request merged**